src/functions/verifyDependencies.js

```diff
@@ -0,0 +1,45 @@
+import generateID from "./generateID";
+
+const verifyDependencies = (tests) => { 
+  if(tests.length > 0) {
+    tests.forEach((test, index) => {
+      const deps = test.deps;
+      const blocks = document.querySelectorAll("*[id^=" + deps + "]");
+      if(blocks.length > 0) {
+        if(test.type === "Source W3CHTML5") {
+          handleFoundDependency("source-comparators", deps, index)
+        } else {
+          handleFoundDependency(generateID(test), deps, index)
+        }
+      } else {
+        handleNotFoundDependency(generateID(test), test.depType, index);
+      }
+    });
+  }
+}
+
+function handleNotFoundDependency(testID, depType, index) {
+  document.querySelectorAll("#" + testID + "-" + index).forEach((block) => {
+    if(depType === "Warning") {
+      block.classList.add("block-warning");
+    } else if(depType === "Error") {
+      block.classList.add("block-error");
+    }
+  });
+}
+
+function handleFoundDependency(testID, depID, index) {
+  document.querySelectorAll("#" + testID + "-" + index).forEach((block) => {
```

Comment:
I would concat both arrays and handle them inside one forEach

Reply:
Done.